dependencies/adi/command_parser/command_parser.cpp

```diff
@@ -65,25 +70,51 @@ void CommandParser::parseArguments(
         }
         if (mandatory) {
             continue;
-        } else if (contains_equal != -1) {
+        } else if (contains_equal != -1) { // Solves -arg/--arg=value
             m_command_vector.push_back(
                 {std::string(argv[i]).substr(0, contains_equal),
                  std::string(argv[i]).substr(contains_equal + 1)});
             arg_number++;
-        } else if (std::string(argv[i]) == "-h" ||
-                   std::string(argv[i]) == "--help") {
-            m_command_vector.push_back({argv[i], "help_menu"});
-            arg_number++;
-        } else if (std::string(argv[i]) == "-s" ||
-                   std::string(argv[i]) == "--split") {
-            m_command_vector.push_back({argv[i], "split"});
-            arg_number++;
-        } else if (i != argc - 1) {
+        } else if (i != argc - 1 && is_argument != 0 &&
+                   i + 1 != argc - 1) { // Solves -arg/--arg value
             m_command_vector.push_back({argv[i], argv[i + 1]});
             i++;
             arg_number += 2;
+        } else if (i != argc - 1 &&
+                   is_argument == 0) { // Solves -arg/--arg -arg value
+            int is_long_arg = std::string(argv[i]).find("--");
+            if (is_long_arg != -1) {
+                bool argument_found = false;
+                for (const auto& entry : command_map) {
+                    if (entry.second.long_option == argv[i]) {
+                        if (!entry.second.has_value) {
+                            m_command_vector.push_back({argv[i], "true"});
+                            arg_number++;
```

Comment:
arg_numer++ and argument_found=true happen on both branches of the if statement

Reply:
Solved